File: packages/contracts-abi/src/addresses.ts
// Per-chain deployed contract addresses, populated by deploy scripts.
// Services and frontend read this rather than hardcoding addresses.
//
// The JSON under contracts/deployments/ is the source of truth; these
// exports merely shape it into a typed object. Regenerate after each deploy.

import type { Address } from "./types.js";

export type ChainName = "kite-testnet" | "base-sepolia" | "arbitrum-sepolia" | "anvil";

export type HeliosAddresses = {
  readonly helios?: Address;
  readonly userVault?: Address;
  readonly allocatorVault?: Address;
  readonly strategyRegistry?: Address;
  readonly allocatorRegistry?: Address;
  readonly reputationAnchor?: Address;
  readonly tradeAttestationVerifier?: Address;
  readonly heliosOApp?: Address;
};

export const CHAIN_IDS: Record<ChainName, number> = {
  "kite-testnet": 2368,
  "base-sepolia": 84_532,
  "arbitrum-sepolia": 421_614,
  "anvil": 31_337,
};

// Populated by `pnpm --filter contracts-abi build` after each deploy.
// The script reads contracts/deployments/*.json and emits this object.
// Phase 0 leaves it empty.
export const ADDRESSES: Record<ChainName, HeliosAddresses> = {
  "kite-testnet": {},
  "base-sepolia": {},
  "arbitrum-sepolia": {},
  "anvil": {},
};

File: packages/contracts-abi-py/src/helios_contracts_abi/addresses.py
"""Per-chain deployed Helios contract addresses.

Populated by the generator after each deploy. Services read from this module
rather than hardcoding addresses, so redeploys only require regeneration.
"""

from typing import Literal, TypedDict

ChainName = Literal["kite-testnet", "base-sepolia", "arbitrum-sepolia", "anvil"]


class HeliosAddresses(TypedDict, total=False):
    helios: str
    userVault: str
    allocatorVault: str
    strategyRegistry: str
    allocatorRegistry: str
    reputationAnchor: str
    tradeAttestationVerifier: str
    heliosOApp: str


CHAIN_IDS: dict[ChainName, int] = {
    "kite-testnet": 2368,
    "base-sepolia": 84_532,
    "arbitrum-sepolia": 421_614,
    "anvil": 31_337,
}

# Phase 0 ships this empty. Populated by deploy scripts + `pnpm build`.
ADDRESSES: dict[ChainName, HeliosAddresses] = {
    "kite-testnet": {},
    "base-sepolia": {},
    "arbitrum-sepolia": {},
    "anvil": {},
}
